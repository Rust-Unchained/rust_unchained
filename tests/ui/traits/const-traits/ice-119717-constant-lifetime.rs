#![allow(incomplete_features)]
#![feature(const_trait_impl, effects, try_trait_v2)]

use std::ops::FromResidual;

impl<T> const FromResidual for T { //~ E0119
    //~^ ERROR const `impl` for trait `FromResidual` which is not marked with `#[const_trait]`
    fn from_residual(t: T) -> _ {
        //~^ the placeholder `_` is not allowed
        t
    }
}

fn main() {}
