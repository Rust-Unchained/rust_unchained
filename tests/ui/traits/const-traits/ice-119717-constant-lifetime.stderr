error: const `impl` for trait `FromResidual` which is not marked with `#[const_trait]`
  --> $DIR/ice-119717-constant-lifetime.rs:6:15
   |
LL | impl<T> const FromResidual for T {
   |               ^^^^^^^^^^^^ this trait is not `const`
   |
   = note: marking a trait with `#[const_trait]` ensures all default method bodies are `const`
   = note: adding a non-const method body in the future would be a breaking change

error[E0119]: conflicting implementations of trait `FromResidual<Result<Infallible, _>>` for type `Result<_, _>`
  --> $DIR/ice-119717-constant-lifetime.rs:6:1
   |
LL | impl<T> const FromResidual for T {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T, E, F> FromResidual<Result<Infallible, E>> for Result<T, F>
             where F: From<E>;

error[E0121]: the placeholder `_` is not allowed within types on item signatures for functions
  --> $DIR/ice-119717-constant-lifetime.rs:8:31
   |
LL |     fn from_residual(t: T) -> _ {
   |                               ^ not allowed in type signatures
   |
help: try replacing `_` with the type in the corresponding trait method signature
   |
LL -     fn from_residual(t: T) -> _ {
LL +     fn from_residual(t: T) -> T {
   |

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0119, E0121.
For more information about an error, try `rustc --explain E0119`.
